contracts/FlowIDTableStaking.cdc

```diff
@@ -870,6 +870,79 @@ pub contract FlowIDTableStaking {
             }
         }
 
+        /// Set slot (count) limits for each node type (role).
+        pub fun setSlotLimits(slotLimits: {UInt8: UInt8}) {
+            FlowIDTableStaking.account.load<{UInt8: UInt8}>(from: /storage/flowStakingSlotLimits)
+            FlowIDTableStaking.account.save(slotLimits, to: /storage/flowStakingSlotLimits)
+        }
+
+        /// Enforces slot (count) limits for each node type (role). If more nodes
+        /// of a given type are registered than the limit, they are removed
+        /// randomly until the limits are met. When a node is removed, it's
+        /// committed tokens are transfered to it's unstaked bucket.
+        pub fun slotSelect() {
+            let allNodeIDs = FlowIDTableStaking.getNodeIDs()
+
+            let slotLimits = FlowIDTableStaking.account.load<{UInt8: UInt8}>(from: /storage/flowStakingSlotLimits)
+                ?? panic("Could not load the slot limits object from storage!")
+
+            // count the nodes of each type
+            var nodeCounts: {UInt8: UInt8} = {1: 0, 2: 0, 3: 0, 4: 0, 5: 0}
+            for nodeID in allNodeIDs {
+                let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
+                if nodeRecord.initialWeight == 100 {
```

Comment:
We can't create new contract fields any more, but we could store the constants in storage and retrieve them with a getter function like we do with `getStakingEnabled()` if we want. Do you think we should do that?

Reply:
Constants for node roles would be nice but I believe that is fine to punt on since a future staking auction system could handle it.

As for defaults, I think they will be set during the initial contract upgrade and that should be fine as it will be documented during the upgrade process.

Reply:
Yeah, that's what I remember - if we can't add "real" constants (`let x = ...`) feel free to disregard this suggestion